File: android_app/app/src/main/java/com/filescanner/app/data/database/entity/DuplicateRow.kt
package com.filescanner.app.data.database.entity

import androidx.room.ColumnInfo

/**
 * 复刻 PC 端"勾选重复"逻辑所需的轻量投影：
 * id / 文件名(fileName) / 书名(title) / 作者(author) / 进度(progress) / 来源(source) / 大小(fileSize) / 创建时间(createdAt)。
 */
data class DuplicateRow(
    val id: Long,
    @ColumnInfo(name = "fileName") val fileName: String,
    val title: String,
    val author: String,
    val progress: String,
    @ColumnInfo(name = "source") val source: String = "",
    @ColumnInfo(name = "fileSize") val fileSize: Long,
    @ColumnInfo(name = "createdAt") val createdAt: Long
)

File: harmony_app/entry/src/main/ets/model/DuplicateRow.ts
/**
 * 复刻“勾选重复”逻辑所需的轻量投影，对齐安卓端 DuplicateRow。
 * id / fileName / title / author / progress / fileSize / createdAt / fileDate / contentHash。
 * contentHash 用于「rule_hash 内容哈希去重」规则：哈希相同且非最新的文件额外勾选。
 */
export class DuplicateRow {
  id: number = 0;
  fileName: string = '';
  title: string = '';
  author: string = '';
  progress: string = '';
  source: string = '';
  fileSize: number = 0;
  createdAt: number = 0;
  /** 文件在文件系统中的最后修改时间（毫秒时间戳）。判断“新旧”时优先使用，避免扫描入库顺序干扰。 */
  fileDate: number = 0;
  /** 内容哈希（扫描时计算，可能为空）。rule_hash 内容哈希去重规则使用。 */
  contentHash: string = '';
}
